### ISSUE TITLE
Crashes with segmentation fault

### ISSUE THREAD

**Issue by SLeitgeb** [Zebradil/xkb-switch-i3#6]
I'm using this program to switch between my two custom layouts by running `xkb-switch --i3` and also in a screen locking script:
```bash
LAYOUT=$(xkb-switch)
xkb-switch -s 'cz(cz_programmer)'
i3lock-fancy-rapid 5 3 -f -n
xkb-switch -s $LAYOUT
```

I used it this way for quite some time with no issues, but recently started to notice random crashes (around once a day, sometimes more frequently). This happens on both my desktop and laptop running Arch Linux. So far I haven't been able to notice if this happens when switching windows / on suspend or resume, etc.

Journal output with some context below. Any recommendations on further debugging?

```
Feb 20 20:58:14 says rtkit-daemon[809]: Supervising 8 threads of 3 processes of 1 users.
Feb 20 20:58:14 says rtkit-daemon[809]: Supervising 8 threads of 3 processes of 1 users.
Feb 20 20:58:14 says rtkit-daemon[809]: Supervising 8 threads of 3 processes of 1 users.
Feb 20 20:58:14 says rtkit-daemon[809]: Successfully made thread 316483 of process 316483 owned by '1000' high priority at nice level -10.
Feb 20 20:58:14 says rtkit-daemon[809]: Supervising 9 threads of 4 processes of 1 users.
Feb 20 20:58:14 says rtkit-daemon[809]: Supervising 9 threads of 4 processes of 1 users.
Feb 20 20:58:14 says rtkit-daemon[809]: Supervising 9 threads of 4 processes of 1 users.
Feb 20 20:58:14 says rtkit-daemon[809]: Supervising 9 threads of 4 processes of 1 users.
Feb 20 20:58:14 says rtkit-daemon[809]: Successfully made thread 316577 of process 316577 owned by '1000' high priority at nice level -10.
Feb 20 20:58:14 says rtkit-daemon[809]: Supervising 10 threads of 5 processes of 1 users.
Feb 20 20:59:52 says audit[772]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=1 pid=772 comm="xkb-switch" exe="/usr/bin/xkb-switch" sig=11 res=1
Feb 20 20:59:52 says kernel: xkb-switch[772]: segfault at 90 ip 00007f5a71aea3e0 sp 00007fff41a9b950 error 4 in libX11.so.6.4.0[7f5a71a63000+8a000]
Feb 20 20:59:52 says kernel: Code: 85 ff 74 06 ff 15 10 96 09 00 48 83 c4 18 48 89 ef 5b 5d 41 5c 41 5d 41 5e 41 5f ff 25 f9 95 09 00 66 0f 1f 84 00 00 00 00 00 <48> 83 7f 40 00 0f 84 c5 00 00 00 41 bc 20 00 00 00 41 bd 10 00 00
Feb 20 20:59:52 says kernel: audit: type=1701 audit(1645387192.513:416): auid=1000 uid=1000 gid=1000 ses=1 pid=772 comm="xkb-switch" exe="/usr/bin/xkb-switch" sig=11 res=1
Feb 20 20:59:52 says audit: BPF prog-id=72 op=LOAD
Feb 20 20:59:52 says audit: BPF prog-id=73 op=LOAD
Feb 20 20:59:52 says audit: BPF prog-id=74 op=LOAD
Feb 20 20:59:52 says systemd[1]: Started Process Core Dump (PID 317206/UID 0).
Feb 20 20:59:52 says audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@2-317206-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 20 20:59:52 says kernel: audit: type=1334 audit(1645387192.517:417): prog-id=72 op=LOAD
Feb 20 20:59:52 says kernel: audit: type=1334 audit(1645387192.517:418): prog-id=73 op=LOAD
Feb 20 20:59:52 says kernel: audit: type=1334 audit(1645387192.517:419): prog-id=74 op=LOAD
Feb 20 20:59:52 says kernel: audit: type=1130 audit(1645387192.517:420): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@2-317206-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 20 20:59:52 says systemd-coredump[317207]: Process 772 (xkb-switch) of user 1000 dumped core.
                                               
                                               Module linux-vdso.so.1 with build-id e70f128dc93a0ce4141b994b2bf8b52a8ba40f28
                                               Module libXdmcp.so.6 with build-id 8ca0792d23c8b8b4c0864297512349292bea5955
                                               Module libXau.so.6 with build-id 1c67764663e07bec24d8951e5fd93f4d165979ff
                                               Module libdl.so.2 with build-id bb9bd2657bfba9f60bd34d2050cc63a7eb024bc4
                                               Module libxcb.so.1 with build-id 0d1ef11740a5daad2ee331e812a51aa6574af222
                                               Module ld-linux-x86-64.so.2 with build-id c09c6f50f6bcec73c64a0b4be77eadb8f7202410
                                               Module libm.so.6 with build-id 596b63a006a4386dcab30912d2b54a7a61827b07
                                               Module libxkbfile.so.1 with build-id de5e415879c20263a8dd42f8cf2dd88107c2bfb2
                                               Module libX11.so.6 with build-id 5ba5798d193c0065014b8c6252a0678671c8d478
                                               Module libc.so.6 with build-id 85766e9d8458b16e9c7ce6e07c712c02b8471dbc
                                               Module libgcc_s.so.1 with build-id 5d817452a709ca3a213341555ddcf446ecee37fa
                                               Module libstdc++.so.6 with build-id 88ad4eff81a00c684abfe0f863e87434123d8943
                                               Module libjsoncpp.so.25 with build-id 87680bad0ee329d980627591e5543fef570aa048
                                               Module libsigc-2.0.so.0 with build-id 6b1f9f8e4d7ad41a0a5ac1c861dbab8616010a32
                                               Module libxkbswitch.so.1 with build-id 7b2d0b8fe5838d6fac36ba4bad63604a5181fddf
                                               Module xkb-switch with build-id 014a4a3b52c2e57415815b864069f459c62be918
                                               Stack trace of thread 772:
                                               #0  0x00007f5a71aea3e0 XkbFreeGeometry (libX11.so.6 + 0xa33e0)
                                               #1  0x00007f5a71aebff1 XkbFreeKeyboard (libX11.so.6 + 0xa4ff1)
                                               #2  0x00007f5a7201b32e _ZN2kb9XKeyboardD1Ev (libxkbswitch.so.1 + 0x432e)
                                               #3  0x0000559226ae51db n/a (xkb-switch + 0x91db)
                                               #4  0x00007f5a71bb8310 __libc_start_call_main (libc.so.6 + 0x2d310)
                                               #5  0x00007f5a71bb83c1 __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x2d3c1)
                                               #6  0x0000559226aea38e _start (xkb-switch + 0xe38e)
                                               ELF object binary architecture: AMD x86-64
Feb 20 20:59:52 says kernel: audit: type=1131 audit(1645387192.550:421): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@2-317206-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 20 20:59:52 says audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@2-317206-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 20 20:59:52 says systemd[1]: systemd-coredump@2-317206-0.service: Deactivated successfully.
Feb 20 20:59:52 says audit: BPF prog-id=0 op=UNLOAD
Feb 20 20:59:52 says audit: BPF prog-id=0 op=UNLOAD
Feb 20 20:59:52 says audit: BPF prog-id=0 op=UNLOAD
Feb 20 20:59:52 says kernel: audit: type=1334 audit(1645387192.627:422): prog-id=0 op=UNLOAD
Feb 20 20:59:52 says kernel: audit: type=1334 audit(1645387192.627:423): prog-id=0 op=UNLOAD
Feb 20 20:59:52 says kernel: audit: type=1334 audit(1645387192.627:424): prog-id=0 op=UNLOAD
--
Feb 22 22:22:10 says bluetoothd[537]: src/profile.c:record_cb() Unable to get Hands-Free Voice gateway SDP record: Host is down
Feb 22 22:22:36 says systemd[1]: systemd-hostnamed.service: Deactivated successfully.
Feb 22 22:22:36 says audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-hostnamed comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 22 22:22:36 says kernel: audit: type=1131 audit(1645564956.506:766): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-hostnamed comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 22 22:22:36 says audit: BPF prog-id=0 op=UNLOAD
Feb 22 22:22:36 says audit: BPF prog-id=0 op=UNLOAD
Feb 22 22:22:36 says audit: BPF prog-id=0 op=UNLOAD
Feb 22 22:22:36 says kernel: audit: type=1334 audit(1645564956.633:767): prog-id=0 op=UNLOAD
Feb 22 22:22:36 says kernel: audit: type=1334 audit(1645564956.633:768): prog-id=0 op=UNLOAD
Feb 22 22:22:36 says kernel: audit: type=1334 audit(1645564956.633:769): prog-id=0 op=UNLOAD
Feb 22 22:26:19 says audit[458488]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=1 pid=458488 comm="xkb-switch" exe="/usr/bin/xkb-switch" sig=11 res=1
Feb 22 22:26:19 says kernel: xkb-switch[458488]: segfault at 90 ip 00007fcfd639e3e0 sp 00007ffc514fdaf0 error 4 in libX11.so.6.4.0[7fcfd6317000+8a000]
Feb 22 22:26:19 says kernel: Code: 85 ff 74 06 ff 15 10 96 09 00 48 83 c4 18 48 89 ef 5b 5d 41 5c 41 5d 41 5e 41 5f ff 25 f9 95 09 00 66 0f 1f 84 00 00 00 00 00 <48> 83 7f 40 00 0f 84 c5 00 00 00 41 bc 20 00 00 00 41 bd 10 00 00
Feb 22 22:26:19 says kernel: audit: type=1701 audit(1645565179.249:770): auid=1000 uid=1000 gid=1000 ses=1 pid=458488 comm="xkb-switch" exe="/usr/bin/xkb-switch" sig=11 res=1
Feb 22 22:26:19 says audit: BPF prog-id=108 op=LOAD
Feb 22 22:26:19 says audit: BPF prog-id=109 op=LOAD
Feb 22 22:26:19 says audit: BPF prog-id=110 op=LOAD
Feb 22 22:26:19 says systemd[1]: Started Process Core Dump (PID 784530/UID 0).
Feb 22 22:26:19 says audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@8-784530-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 22 22:26:19 says kernel: audit: type=1334 audit(1645565179.253:771): prog-id=108 op=LOAD
Feb 22 22:26:19 says kernel: audit: type=1334 audit(1645565179.253:772): prog-id=109 op=LOAD
Feb 22 22:26:19 says kernel: audit: type=1334 audit(1645565179.253:773): prog-id=110 op=LOAD
Feb 22 22:26:19 says kernel: audit: type=1130 audit(1645565179.253:774): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@8-784530-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 22 22:26:19 says systemd-coredump[784531]: Process 458488 (xkb-switch) of user 1000 dumped core.
                                               
                                               Module linux-vdso.so.1 with build-id e70f128dc93a0ce4141b994b2bf8b52a8ba40f28
                                               Module libXdmcp.so.6 with build-id 8ca0792d23c8b8b4c0864297512349292bea5955
                                               Module libXau.so.6 with build-id 1c67764663e07bec24d8951e5fd93f4d165979ff
                                               Module libdl.so.2 with build-id bb9bd2657bfba9f60bd34d2050cc63a7eb024bc4
                                               Module libxcb.so.1 with build-id 0d1ef11740a5daad2ee331e812a51aa6574af222
                                               Module ld-linux-x86-64.so.2 with build-id c09c6f50f6bcec73c64a0b4be77eadb8f7202410
                                               Module libm.so.6 with build-id 596b63a006a4386dcab30912d2b54a7a61827b07
                                               Module libxkbfile.so.1 with build-id de5e415879c20263a8dd42f8cf2dd88107c2bfb2
                                               Module libX11.so.6 with build-id 5ba5798d193c0065014b8c6252a0678671c8d478
                                               Module libc.so.6 with build-id 85766e9d8458b16e9c7ce6e07c712c02b8471dbc
                                               Module libgcc_s.so.1 with build-id 5d817452a709ca3a213341555ddcf446ecee37fa
                                               Module libstdc++.so.6 with build-id 88ad4eff81a00c684abfe0f863e87434123d8943
                                               Module libjsoncpp.so.25 with build-id 87680bad0ee329d980627591e5543fef570aa048
                                               Module libsigc-2.0.so.0 with build-id 6b1f9f8e4d7ad41a0a5ac1c861dbab8616010a32
                                               Module libxkbswitch.so.1 with build-id 7b2d0b8fe5838d6fac36ba4bad63604a5181fddf
                                               Module xkb-switch with build-id 014a4a3b52c2e57415815b864069f459c62be918
                                               Stack trace of thread 458488:
                                               #0  0x00007fcfd639e3e0 XkbFreeGeometry (libX11.so.6 + 0xa33e0)
                                               #1  0x00007fcfd639fff1 XkbFreeKeyboard (libX11.so.6 + 0xa4ff1)
                                               #2  0x00007fcfd68cf32e _ZN2kb9XKeyboardD1Ev (libxkbswitch.so.1 + 0x432e)
                                               #3  0x000055a7f676f1db n/a (xkb-switch + 0x91db)
                                               #4  0x00007fcfd646c310 __libc_start_call_main (libc.so.6 + 0x2d310)
                                               #5  0x00007fcfd646c3c1 __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x2d3c1)
                                               #6  0x000055a7f677438e _start (xkb-switch + 0xe38e)
                                               ELF object binary architecture: AMD x86-64
Feb 22 22:26:19 says audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@8-784530-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 22 22:26:19 says kernel: audit: type=1131 audit(1645565179.286:775): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@8-784530-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 22 22:26:19 says systemd[1]: systemd-coredump@8-784530-0.service: Deactivated successfully.
Feb 22 22:26:19 says audit: BPF prog-id=0 op=UNLOAD
Feb 22 22:26:19 says audit: BPF prog-id=0 op=UNLOAD
Feb 22 22:26:19 says audit: BPF prog-id=0 op=UNLOAD
Feb 22 22:26:19 says kernel: audit: type=1334 audit(1645565179.393:776): prog-id=0 op=UNLOAD
Feb 22 22:26:19 says kernel: audit: type=1334 audit(1645565179.393:777): prog-id=0 op=UNLOAD
Feb 22 22:26:19 says kernel: audit: type=1334 audit(1645565179.393:778): prog-id=0 op=UNLOAD
--
Feb 25 17:25:04 says kernel: audit: type=1130 audit(1645806304.405:1164): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-hostnamed comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 25 17:25:46 says systemd[1]: systemd-hostnamed.service: Deactivated successfully.
Feb 25 17:25:46 says audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-hostnamed comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 25 17:25:46 says kernel: audit: type=1131 audit(1645806346.195:1165): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-hostnamed comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 25 17:25:46 says audit: BPF prog-id=0 op=UNLOAD
Feb 25 17:25:46 says audit: BPF prog-id=0 op=UNLOAD
Feb 25 17:25:46 says audit: BPF prog-id=0 op=UNLOAD
Feb 25 17:25:46 says kernel: audit: type=1334 audit(1645806346.315:1166): prog-id=0 op=UNLOAD
Feb 25 17:25:46 says kernel: audit: type=1334 audit(1645806346.315:1167): prog-id=0 op=UNLOAD
Feb 25 17:25:46 says kernel: audit: type=1334 audit(1645806346.315:1168): prog-id=0 op=UNLOAD
Feb 25 17:26:39 says audit[1101426]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=1 pid=1101426 comm="xkb-switch" exe="/usr/bin/xkb-switch" sig=11 res=1
Feb 25 17:26:39 says kernel: xkb-switch[1101426]: segfault at 90 ip 00007ff7ca6573e0 sp 00007ffe1b070110 error 4 in libX11.so.6.4.0[7ff7ca5d0000+8a000]
Feb 25 17:26:39 says kernel: Code: 85 ff 74 06 ff 15 10 96 09 00 48 83 c4 18 48 89 ef 5b 5d 41 5c 41 5d 41 5e 41 5f ff 25 f9 95 09 00 66 0f 1f 84 00 00 00 00 00 <48> 83 7f 40 00 0f 84 c5 00 00 00 41 bc 20 00 00 00 41 bd 10 00 00
Feb 25 17:26:39 says kernel: audit: type=1701 audit(1645806399.965:1169): auid=1000 uid=1000 gid=1000 ses=1 pid=1101426 comm="xkb-switch" exe="/usr/bin/xkb-switch" sig=11 res=1
Feb 25 17:26:39 says audit: BPF prog-id=144 op=LOAD
Feb 25 17:26:39 says audit: BPF prog-id=145 op=LOAD
Feb 25 17:26:39 says audit: BPF prog-id=146 op=LOAD
Feb 25 17:26:39 says systemd[1]: Started Process Core Dump (PID 1290985/UID 0).
Feb 25 17:26:39 says kernel: audit: type=1334 audit(1645806399.968:1170): prog-id=144 op=LOAD
Feb 25 17:26:39 says kernel: audit: type=1334 audit(1645806399.968:1171): prog-id=145 op=LOAD
Feb 25 17:26:39 says kernel: audit: type=1334 audit(1645806399.968:1172): prog-id=146 op=LOAD
Feb 25 17:26:39 says kernel: audit: type=1130 audit(1645806399.972:1173): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@9-1290985-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 25 17:26:39 says audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@9-1290985-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 25 17:26:40 says systemd-coredump[1290986]: Process 1101426 (xkb-switch) of user 1000 dumped core.
                                                
                                                Module linux-vdso.so.1 with build-id e70f128dc93a0ce4141b994b2bf8b52a8ba40f28
                                                Module libXdmcp.so.6 with build-id 8ca0792d23c8b8b4c0864297512349292bea5955
                                                Module libXau.so.6 with build-id 1c67764663e07bec24d8951e5fd93f4d165979ff
                                                Module libdl.so.2 with build-id bb9bd2657bfba9f60bd34d2050cc63a7eb024bc4
                                                Module libxcb.so.1 with build-id 0d1ef11740a5daad2ee331e812a51aa6574af222
                                                Module ld-linux-x86-64.so.2 with build-id c09c6f50f6bcec73c64a0b4be77eadb8f7202410
                                                Module libm.so.6 with build-id 596b63a006a4386dcab30912d2b54a7a61827b07
                                                Module libxkbfile.so.1 with build-id de5e415879c20263a8dd42f8cf2dd88107c2bfb2
                                                Module libX11.so.6 with build-id 5ba5798d193c0065014b8c6252a0678671c8d478
                                                Module libc.so.6 with build-id 85766e9d8458b16e9c7ce6e07c712c02b8471dbc
                                                Module libgcc_s.so.1 with build-id 5d817452a709ca3a213341555ddcf446ecee37fa
                                                Module libstdc++.so.6 with build-id 88ad4eff81a00c684abfe0f863e87434123d8943
                                                Module libjsoncpp.so.25 with build-id 87680bad0ee329d980627591e5543fef570aa048
                                                Module libsigc-2.0.so.0 with build-id 6b1f9f8e4d7ad41a0a5ac1c861dbab8616010a32
                                                Module libxkbswitch.so.1 with build-id 7b2d0b8fe5838d6fac36ba4bad63604a5181fddf
                                                Module xkb-switch with build-id 014a4a3b52c2e57415815b864069f459c62be918
                                                Stack trace of thread 1101426:
                                                #0  0x00007ff7ca6573e0 XkbFreeGeometry (libX11.so.6 + 0xa33e0)
                                                #1  0x00007ff7ca658ff1 XkbFreeKeyboard (libX11.so.6 + 0xa4ff1)
                                                #2  0x00007ff7cab8832e _ZN2kb9XKeyboardD1Ev (libxkbswitch.so.1 + 0x432e)
                                                #3  0x00005643120361db n/a (xkb-switch + 0x91db)
                                                #4  0x00007ff7ca725310 __libc_start_call_main (libc.so.6 + 0x2d310)
                                                #5  0x00007ff7ca7253c1 __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x2d3c1)
                                                #6  0x000056431203b38e _start (xkb-switch + 0xe38e)
                                                ELF object binary architecture: AMD x86-64
Feb 25 17:26:40 says kernel: audit: type=1131 audit(1645806399.998:1174): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@9-1290985-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 25 17:26:39 says audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@9-1290985-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Feb 25 17:26:40 says systemd[1]: systemd-coredump@9-1290985-0.service: Deactivated successfully.
Feb 25 17:26:40 says audit: BPF prog-id=0 op=UNLOAD
Feb 25 17:26:40 says audit: BPF prog-id=0 op=UNLOAD
Feb 25 17:26:40 says audit: BPF prog-id=0 op=UNLOAD
Feb 25 17:26:40 says kernel: audit: type=1334 audit(1645806400.075:1175): prog-id=0 op=UNLOAD
Feb 25 17:26:40 says kernel: audit: type=1334 audit(1645806400.075:1176): prog-id=0 op=UNLOAD
Feb 25 17:26:40 says kernel: audit: type=1334 audit(1645806400.075:1177): prog-id=0 op=UNLOAD
```

**Comment by SLeitgeb** [comment]
Finally got a crash when running `xkb-switch --i3` manually in a shell.

The crash message:
`Job 1, 'xkb-switch --i3' terminated by signal SIGSEGV (Address boundary error)`

**Comment by Zebradil** [comment]
Thanks for reporting and sorry for the delay. Notifications for this repo were turned off.

Please, check the latest version (1.8.5+i3-2). The recently merged PR from @akorb should fix the issue.

**Comment by SLeitgeb** [comment]
Thanks for the effort. Now I get crashes with a different error on both my machines:
```LargestInt out of UInt range```

Should I open a new issue or provide more info?

**Comment by Zebradil** [comment]
@SLeitgeb, it is fixed in the latest version ([1.8.5+i3-4](https://github.com/Zebradil/xkb-switch-i3/releases/tag/1.8.5%2Bi3-4), try it out.
